python/foglamp/services/core/api/service.py

```diff
@@ -110,17 +111,24 @@ def get_service_records():
         try:
             # "plugin_module_path" is fixed by design. It is MANDATORY to keep the plugin in the exactly similar named
             # folder, within the plugin_module_path.
+            # if multiple plugin with same name are found, then python plugin import will be tried first
             plugin_module_path = "foglamp.plugins.south" if service_type == 'south' else "foglamp.plugins.north"
             import_file_name = "{path}.{dir}.{file}".format(path=plugin_module_path, dir=plugin, file=plugin)
             _plugin = __import__(import_file_name, fromlist=[''])
 
+            script = '["services/south"]' if service_type == 'south' else '["services/north"]'
             # Fetch configuration from the configuration defined in the plugin
             plugin_info = _plugin.plugin_info()
             plugin_config = plugin_info['config']
         except ImportError as ex:
-            raise web.HTTPNotFound(reason='Plugin "{}" import problem from path "{}". {}'.format(plugin, plugin_module_path, str(ex)))
+            # Checking for C-type plugins
+            script = '["services/south_c"]' if service_type == 'south' else '["services/north_c"]'
+            plugin_info = apiutils.get_plugin_info(plugin)
```

Comment:
@MarkRiddoch See the below scenario
![south_wizard](https://user-images.githubusercontent.com/27723735/43720305-c27b0b4e-99ad-11e8-9367-9fd8a1c6a605.png)

With above configuration, plugin is not loaded after enable and start the schedule.
```
Aug  6 19:19:32 nerd-034 C-dumy[6956]: Load south plugin foglamp-dummy.
Aug  6 19:19:32 nerd-034 C-dumy[6956]: PluginManager: Failed to load plugin foglamp-dummy.
Aug  6 19:19:32 nerd-034 C-dumy[6956]: Failed to load south plugin
```
To make it work, I changed the configuration value to "dummy", because C-south plugin requires same name (i.e dummy) as associated library i.e "lib<dummy>.so"

How would user know to change it like above?

Reply:
@MarkRiddoch 
Having this field as editable could allow the user to set plugin name different than the real lib<plugin_name>.so file present in plugins directory. And that would basically break this flow. Even this value "foglamp-dummy" is not correct to be added in the configuration DB table entry for this plugin.

Is there a reason to have prefix "foglamp-" in the plugin default value in plugin.cpp? Can we just have this value same as library name's middle part? If this is agreeable, probably we can insert this plugin's default name as ${PROJECT_NAME} (i.e. library name's middle part) from C plugin's CMakeLists.txt file.